apiserver/facades/agent/uniter/service.go

```diff
@@ -78,6 +92,7 @@ type UnitStateService interface {
 // PortService describes the ability to open and close port ranges for units.
 type PortService interface {
 	UpdateUnitPorts(ctx context.Context, unitUUID coreunit.UUID, openPorts, closePorts network.GroupedPortRanges) error
+	GetMachineOpenedPorts(ctx context.Context, machineUUID string) (map[coreunit.UUID]network.GroupedPortRanges, error)
```

Comment:
You added comments to the application service indirection, then neglected them here.